packages/aws-cdk-lib/custom-resources/test/provider-framework/waiter-state-machine.test.ts

```diff
@@ -78,6 +104,77 @@ describe('state machine', () => {
         Version: '2012-10-17',
       },
     });
+    Template.fromStack(stack).hasResourceProperties('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: 'lambda:InvokeFunction',
+            Effect: 'Allow',
+            Resource: stack.resolve(isCompleteHandler.resourceArnsForGrantInvoke),
+          },
+          {
+            Action: 'lambda:InvokeFunction',
+            Effect: 'Allow',
+            Resource: stack.resolve(timeoutHandler.resourceArnsForGrantInvoke),
+          },
+          {
+            Action: [
+              'logs:CreateLogDelivery',
+              'logs:CreateLogStream',
+              'logs:GetLogDelivery',
+              'logs:UpdateLogDelivery',
+              'logs:DeleteLogDelivery',
+              'logs:ListLogDeliveries',
+              'logs:PutLogEvents',
+              'logs:PutResourcePolicy',
+              'logs:DescribeResourcePolicies',
+              'logs:DescribeLogGroups',
+            ],
+            Effect: 'Allow',
+            Resource: '*',
+          },
+        ],
+        Version: '2012-10-17',
+      },
+      Roles: [{ Ref: roleId }],
+    });
+    Template.fromStack(stack).hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: 1,
+    });
+  });
+
+  test('disable logging', () => {
+    // GIVEN
+    const stack = new Stack();
+    Node.of(stack).setContext('@aws-cdk/core:target-partitions', ['aws', 'aws-cn']);
+
+    const isCompleteHandler = new lambdaFn(stack, 'isComplete', {
+      code: Code.fromInline('foo'),
+      runtime: lambda.Runtime.NODEJS_LATEST,
+      handler: 'index.handler',
+    });
+    const timeoutHandler = new lambdaFn(stack, 'isTimeout', {
+      code: Code.fromInline('foo'),
+      runtime: lambda.Runtime.NODEJS_LATEST,
+      handler: 'index.handler',
+    });
+    const interval = Duration.hours(2);
+    const maxAttempts = 2;
+    const backoffRate = 5;
+
+    // WHEN
+    new WaiterStateMachine(stack, 'statemachine', {
+      isCompleteHandler,
+      timeoutHandler,
+      backoffRate,
+      interval,
+      maxAttempts,
+      disableLogging: true,
+    });
+
+    // THEN
+    const roleId = 'statemachineRole52044F93';
+    Template.fromStack(stack).resourceCountIs('AWS::Logs::LogGroup', 0);
     Template.fromStack(stack).hasResourceProperties('AWS::IAM::Policy', {
```

Comment:
```suggestion
```
`Template.fromStack(stack).resourceCountIs('AWS::Logs::LogGroup', 0);` is enough for this test case.